src/builder/Actions.sol

```diff
@@ -477,6 +483,99 @@ library Actions {
         return (quarkOperation, action);
     }
 
+    struct BorrowInput {
+        Accounts.ChainAccounts[] chainAccountsList;
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        address borrower;
+        uint256 chainId;
+        uint256[] collateralAmounts;
+        string[] collateralTokenSymbols;
+        address comet;
+    }
+
+    function borrow(BorrowInput memory borrowInput, PaymentInfo.Payment memory payment)
+        internal
+        pure
+        returns (IQuarkWallet.QuarkOperation memory, Action memory)
+    {
+        bytes[] memory scriptSources = new bytes[](1);
+        scriptSources[0] = type(CometSupplyMultipleAssetsAndBorrow).creationCode;
+
+        Accounts.ChainAccounts memory accounts =
+            Accounts.findChainAccounts(borrowInput.chainId, borrowInput.chainAccountsList);
+
+        Accounts.QuarkState memory accountState = Accounts.findQuarkState(borrowInput.borrower, accounts.quarkStates);
+
+        Accounts.AssetPositions memory borrowAssetPositions =
+            Accounts.findAssetPositions(borrowInput.assetSymbol, accounts.assetPositionsList);
+
+        uint256[] memory collateralTokenPrices = new uint256[](borrowInput.collateralTokenSymbols.length);
+        address[] memory collateralTokens = new address[](borrowInput.collateralTokenSymbols.length);
+
+        for (uint256 i = 0; i < borrowInput.collateralTokenSymbols.length; ++i) {
+            Accounts.AssetPositions memory assetPositions =
+                Accounts.findAssetPositions(borrowInput.collateralTokenSymbols[i], accounts.assetPositionsList);
+            collateralTokenPrices[i] = assetPositions.usdPrice;
+            collateralTokens[i] = assetPositions.asset;
+        }
+
+        // XXX handle wrapping ETH?
+        bytes memory scriptCalldata = abi.encodeWithSelector(
+            CometSupplyMultipleAssetsAndBorrow.run.selector,
+            borrowInput.comet,
+            collateralTokens,
+            borrowInput.collateralAmounts,
+            borrowAssetPositions.asset,
+            borrowInput.amount
+        );
+
+        // Construct QuarkOperation
+        IQuarkWallet.QuarkOperation memory quarkOperation = IQuarkWallet.QuarkOperation({
+            nonce: accountState.quarkNextNonce,
+            scriptAddress: CodeJarHelper.getCodeAddress(type(CometSupplyMultipleAssetsAndBorrow).creationCode),
+            scriptCalldata: scriptCalldata,
+            scriptSources: scriptSources,
+            expiry: borrowInput.blockTimestamp + STANDARD_EXPIRY_BUFFER
+        });
+
+        if (payment.isToken) {
+            // Wrap operation with paycall
+            quarkOperation = PaycallWrapper.wrap(
+                quarkOperation,
+                borrowInput.chainId,
+                payment.currency,
+                PaymentInfo.findMaxCost(payment, borrowInput.chainId)
+            );
+        }
```

Comment:
Now that we are delaying wrapping paycall until after the operations are merged, this can be removed.

Reply:
removed